src/odemis/acq/acq_stream_settings.py

```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+"""
+Created on 24 Jan 2024
+
+@author: Karishma Kumar
+
+Copyright © 2024-2025 Karishma Kumar, Delmic
+
+This file is part of Odemis.
+
+Odemis is free software: you can redistribute it and/or modify it under the terms of the GNU General Public License version 2 as published by the Free Software Foundation.
+
+Odemis is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with Odemis. If not, see http://www.gnu.org/licenses/.
+"""
+import json
+import logging
+import os
+from datetime import datetime
+from typing import List, Dict, Any
+
+from odemis.gui.util import get_home_folder
+
+CONF_PATH = os.path.join(get_home_folder(), u".config/odemis")
+MAX_LENGTH = 10  # Maximum number most recent settings to display
+
+
+def get_config_index(config_data: list, name: str) -> int:
+    """
+    Get the index of the config entry with the given name
+    :param config_data: list of config entries
+    :param name: name of config entry
+    :return: index of config entry with given name
+    """
+    try:
+        index = list(map(lambda x: x["name"], config_data)).index(name)
+    except:
+        index = None
+
+    return index
+
+
+def add_and_sort_config_entries(config_data: list, new_entries: list, limit: int = 10) -> List[Dict]:
+    """
+    Add new entries to config_data and sort by last_updated, limit to 10
+    :param config_data: list of config entries
+    :param new_entries: list of new config entries
+    :param limit: limit of config entries
+    :return: sorted list of config entries
+    """
+
+    # add new entries to config_data
+    for entry in new_entries:
+        index = get_config_index(config_data, name=entry["name"])
+        entry["last_updated"] = datetime.now().strftime("%Y-%m-%d %H:%M:%S")
```

Comment:
The idea of storing the settings as a list was that you don't need to store the date. So either you store the whole thing as a dict (stream name -> settings), or you store it as a list, and use the order of the list to drop the oldest ones. 
As you know that the oldest ones are at the end of the list, you can just take the original list, remove the entries which are old version of the new entries, insert the new entries at the beginning, and finally the list to `limit-len(new_entries)`.

Reply:
"As you know that the oldest ones are at the end of the list, you can just take the original list, remove the entries which are old version of the new entries, insert the new entries at the beginning, and finally the list to limit-len(new_entries)." This works nicely when the new filter names are different from the names already stored in the list. When we have a duplication i.e. a same filter name, we search that filter name in the list and move it to top i.e. the most recently used.

Reply:
In my experience, if you try to modify the list at different places, you're easily going to mess up with the index. That is the point of my suggested algorithm: do each change to the list as separate independent steps:
* remove the duplicates from the current list => you're certain the list doesn't contain duplicates
* insert the new entries at the beginning => you're certain all the new entries are at the beginning
* crop => you're certain the list is not too big.

BTW, to insert an element at the beginning of a list you can use `.insert(0, e)`

Reply:
I don't think we know if new_entries is sorted when it is passed in though? depending how it is updated from the ui, mru might not always be last entry in the list, so we need the date to sort. 

Agree, think it would be simpler if it was a dict, and then we can remove get_index, and just update config_data[name] directly.

Reply:
The `config_data` is sorted (from latest used to oldest). The new_entries are not sorted, but the assumption is that they have *just* been used (that's why they are being added to the list). They all have the same time: right now. So the new_entries always go to the beginning of the list.